src/LegacyStrings.jl

```diff
@@ -40,40 +40,43 @@ import Base:
     uppercase,
     write
 
-immutable ASCIIString <: DirectIndexString
-    data::Array{UInt8,1}
-end
-
-immutable UTF8String <: AbstractString
-    data::Vector{UInt8}
-end
+    if VERSION >= v"0.5.0-"
+        immutable ASCIIString <: DirectIndexString
+            data::Array{UInt8,1}
+        end
 
-immutable UTF16String <: AbstractString
-    data::Vector{UInt16} # includes 16-bit NULL termination after string chars
-    function UTF16String(data::Vector{UInt16})
-        if length(data) < 1 || data[end] != 0
-            throw(UnicodeError(UTF_ERR_NULL_16_TERMINATE, 0, 0))
+        immutable UTF8String <: AbstractString
+            data::Vector{UInt8}
```

Comment:
Note I didn't change the code at all. I guess that's just for historical reasons.